fix(kmodal): dismiss icon logic [khcp-4354] (#737)
Kong/kongponents: docs/components/modal.md

Use this to hide the built-in cancel button (`false` by default).

### hideDismissIcon

> When using the `header-image` slot we display a dismiss 'X' button in the upper right corner of the dialog. Boolean for hiding this button (`false` by default).

Reviewer: ```suggestion
When using the `header-image` slot we display a dismiss 'X' button in the upper right corner of the dialog. Set this prop to `true` to hide the button (`false` by default).
```

Author: Ported this change to `beta` pr too
